### PR TITLE
[node] Bring in changes from v18.11.0

### PR DESCRIPTION
**Pull request by meyfa**
See release: https://github.com/nodejs/node/releases/tag/v18.11.0
v18.11.0 brought the following changes:

* `fs/promises`: add `FileHandle.prototype.readLines`
* `http`, `http2`: add `writeEarlyHints()` function
* `util`: add default value option to parsearg
* `util`: add `transferableAbortController()`
* `util`: add `transferableAbortSignal()`
* `crypto`: add null length to `crypto.subtle.deriveBits`
* `path`: change `basename()` argument from ext to suffix

---

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://nodejs.org/api/fs.html#filehandlereadlinesoptions
  - https://nodejs.org/api/http.html#responsewriteearlyhintshints-callback
  - https://nodejs.org/api/http2.html#responsewriteearlyhintslinks
  - https://nodejs.org/api/util.html#utilparseargsconfig
  - https://nodejs.org/api/util.html#utiltransferableabortcontroller
  - https://nodejs.org/api/util.html#utiltransferableabortsignalsignal
  - https://nodejs.org/api/webcrypto.html#subtlederivebitsalgorithm-basekey-length
  - https://nodejs.org/api/path.html#pathbasenamepath-suffix
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/crypto.d.ts [hunk 1]
```diff
@@ -3735,7 +3737,7 @@ declare module 'crypto' {
              * - `'PBKDF2'`
              * @since v15.0.0
              */
-            deriveBits(algorithm: AlgorithmIdentifier | EcdhKeyDeriveParams | HkdfParams | Pbkdf2Params, baseKey: CryptoKey, length: number): Promise<ArrayBuffer>;
+            deriveBits(algorithm: AlgorithmIdentifier | EcdhKeyDeriveParams | HkdfParams | Pbkdf2Params, baseKey: CryptoKey, length: number | null): Promise<ArrayBuffer>;
```
Comment (panva) [inline types/node/crypto.d.ts:3740]:
```suggestion
            deriveBits(algorithm: EcdhKeyDeriveParams, baseKey: CryptoKey, length: number | null): Promise<ArrayBuffer>;
            deriveBits(algorithm: AlgorithmIdentifier | HkdfParams | Pbkdf2Params, baseKey: CryptoKey, length: number): Promise<ArrayBuffer>;
```

File: types/node/test/crypto.ts [hunk 2]
```diff
@@ -1431,6 +1431,7 @@ import { promisify } from 'node:util';
     subtle.encrypt({ name: 'AES-CBC', iv: new Uint8Array(16) }, key, new TextEncoder().encode('hello')); // $ExpectType Promise<ArrayBuffer>
     subtle.decrypt({ name: 'AES-CBC', iv: new Uint8Array(16) }, key, new ArrayBuffer(8)); // $ExpectType Promise<ArrayBuffer>
     subtle.deriveBits({ name: 'PBKDF2', hash: 'SHA-512', salt: new ArrayBuffer(8), iterations: 1000 }, key, length); // $ExpectType Promise<ArrayBuffer>
+    subtle.deriveBits({ name: 'PBKDF2', hash: 'SHA-512', salt: new ArrayBuffer(8), iterations: 1000 }, key, null); // $ExpectType Promise<ArrayBuffer>
```
Comment (panva) [inline types/node/test/crypto.ts:1434]:
```suggestion
    subtle.deriveBits({ name: 'ECDH', public: key }, key, null); // $ExpectType Promise<ArrayBuffer>
```

**Review by panva — CHANGES_REQUESTED** [review]

**Comment by meyfa** [comment]
@panva Nice suggestion, thanks!

**Review by panva — COMMENTED** [review]
crypto changes lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Review by peterblazejewicz — APPROVED** [review]

**Comment by meyfa** [comment]
Ready to merge

**Pull request merged**